Automattic/woocommerce-payments
# Add `is_test_mode` to every JS track event

#### Changes proposed in this Pull Request
Add `is_test_mode` to every JS track event, updating `recordEvent` method to include the property, getting its value from `wcpaySettings.testMode`.

This will allow us to accurately measure live accounts on Tracks funnels.

#### Testing instructions
Go to any WooPayments page and check that the registered `*_wcpay_*` events have `is_test_mode` property defined.

>[!Note]
> Remember to enable `Allow usage of WooCommerce to be Tracked` in **WooCommerce > Settings > Advanced > WooCommerce.com**.

To check that Tracks events are properly registered, you can use one of the following methods:
- Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` on your site browser console. And look for `verbose` logs.
- Install [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) browser extension and use its log of events.
- Go to [Tracks Live view](https://mc.a8c.com/tracks/live/), filter by any of the allowed methods, and wait patiently (~10min) for events to be registered there.



-------------------


**Post merge**
